test/python/compiler/test_assembler.py

```diff
@@ -641,8 +641,11 @@ def test_assemble_with_delay(self):
         delay_qobj = assemble(delay_schedule, backend)
         validate_qobj_against_schema(delay_qobj)
 
-        self.assertEqual(orig_qobj.experiments[0].to_dict(),
-                         delay_qobj.experiments[0].to_dict())
+        for instr in delay_qobj.experiments[0].to_dict()['instructions']:
+            if instr['name'] == 'delay':
```

Comment:
Does this fail with the old implementation? Does this make sure that there is a delay instruction in the qobj? The docstring for this is "test delay is ignored" -- since delay isn't ignored anymore, this should be updated. It seems like we don't need to compare `delay_qobj` to `orig_qobj` anymore, we just want to verify that the delay instruction we get is what we want. Maybe replace `delay_schedule` with a more simple one by removing `orig_schedule + ` on line 637.

Reply:
Fixed in [231e4bb](https://github.com/Qiskit/qiskit-terra/pull/4233/commits/231e4bb63753df33b86b97e265e8092a0415471c).